imports/plugins/included/shipping-rates/server/hooks/hooks.js

```diff
@@ -34,20 +34,40 @@ function getShippingRates(previousQueryResults, cart) {
       return previousQueryResults;
     }
   }
-  if (!(cart.shipping && cart.shipping[0] && cart.shipping[0].address)) {
+
+  // Verify that we have shipping records
+  if (!cart.shipping || !cart.shipping.length) {
     const errorDetails = {
       requestStatus: "error",
-      shippingProvider: "shippo",
-      message: "The 'shipping' property of this cart is either missing or incomplete."
+      shippingProvider: "flat-rate-shipping",
+      message: "this cart is missing shipping records"
     };
-    rates.push(errorDetails);
-    return [rates, retrialTargets];
+    return [[errorDetails], []];
```

Comment:
I feel like these comments are about the error handling techniques of the shipping method hooks which I am not creating or changing, just correcting the fact that it calls out the wrong shipment method.

Reply:
You were the reviewer on the PR (#2738) that introduced this pattern a few weeks ago and this is the first I've seen of this pattern.

We can punt to a separate issue to fix this, but I'd like to understand what is happening here if you know.

Reply:
I will go through the code and try to add comments to try and make this more clear. I do feel like the general pattern here makes sense and works well. It's a little challenging trying to deal with a series of Hooks that can be extended by any plugin and deal with feedback from each hook on how to proceed (do I retry?, etc.) especially since how they need to communicate is with a record on the cart.

The timing of putting in this change while also dealing with multi-shop was extremely unfortunate however.